contracts/base/Dispatcher.sol

```diff
@@ -31,125 +31,143 @@ abstract contract Dispatcher is Payments, V2SwapRouter, V3SwapRouter, Callbacks
-                } else if (command == Commands.V3_SWAP_EXACT_OUT) {
-                    (address recipient, uint256 amountOut, uint256 amountInMax, bytes memory path, bool payerIsUser) =
-                        abi.decode(inputs, (address, uint256, uint256, bytes, bool));
-                    address payer = payerIsUser ? msg.sender : address(this);
-                    v3SwapExactOutput(recipient.map(), amountOut, amountInMax, path, payer);
-                } else if (command == Commands.PERMIT2_TRANSFER_FROM) {
-                    (address token, address recipient, uint160 amount) = abi.decode(inputs, (address, address, uint160));
-                    permit2TransferFrom(token, msg.sender, recipient, amount);
-                } else if (command == Commands.PERMIT2_PERMIT_BATCH) {
-                    (IAllowanceTransfer.PermitBatch memory permitBatch, bytes memory data) =
-                        abi.decode(inputs, (IAllowanceTransfer.PermitBatch, bytes));
-                    PERMIT2.permit(msg.sender, permitBatch, data);
-                } else if (command == Commands.SWEEP) {
-                    (address token, address recipient, uint256 amountMin) =
-                        abi.decode(inputs, (address, address, uint256));
-                    Payments.sweep(token, recipient.map(), amountMin);
-                } else if (command == Commands.TRANSFER) {
-                    (address token, address recipient, uint256 value) = abi.decode(inputs, (address, address, uint256));
-                    Payments.pay(token, recipient.map(), value);
-                } else if (command == Commands.PAY_PORTION) {
-                    (address token, address recipient, uint256 bips) = abi.decode(inputs, (address, address, uint256));
-                    Payments.payPortion(token, recipient.map(), bips);
-                } else {
-                    revert InvalidCommandType(command);
-                }
+				if (is0to3) {
+					if (command == Commands.V3_SWAP_EXACT_IN) {
+						(address recipient, uint256 amountIn, uint256 amountOutMin, bytes memory path, bool payerIsUser) =
+							abi.decode(inputs, (address, uint256, uint256, bytes, bool));
+						address payer = payerIsUser ? msg.sender : address(this);
+						v3SwapExactInput(recipient.map(), amountIn, amountOutMin, path, payer);
+					} else if (command == Commands.V3_SWAP_EXACT_OUT) {
+						(address recipient, uint256 amountOut, uint256 amountInMax, bytes memory path, bool payerIsUser) =
+							abi.decode(inputs, (address, uint256, uint256, bytes, bool));
+						address payer = payerIsUser ? msg.sender : address(this);
+						v3SwapExactOutput(recipient.map(), amountOut, amountInMax, path, payer);
+					} else if (command == Commands.PERMIT2_TRANSFER_FROM) {
+						(address token, address recipient, uint160 amount) = abi.decode(inputs, (address, address, uint160));
+						permit2TransferFrom(token, msg.sender, recipient, amount);
+					} else { // PERMIT2_PERMIT_BATCH
```

Comment:
I guess in this cases i should leave the `else if` and add a final else with a `revert InvalidCommandType(command)`